gen/xbuild.sh

```diff
@@ -8,51 +8,75 @@
-CGO_ENABLED=0 GOOS=openbsd   GOARCH=arm      go build -ldflags="-s -w -X main.gVersion=$version" && tar czf dist/lf-openbsd-arm.tar.gz     lf --remove-files
-CGO_ENABLED=0 GOOS=openbsd   GOARCH=arm64    go build -ldflags="-s -w -X main.gVersion=$version" && tar czf dist/lf-openbsd-arm64.tar.gz   lf --remove-files
-CGO_ENABLED=0 GOOS=solaris   GOARCH=amd64    go build -ldflags="-s -w -X main.gVersion=$version" && tar czf dist/lf-solaris-amd64.tar.gz   lf --remove-files
+ERRORS=
 
-CGO_ENABLED=0 GOOS=windows   GOARCH=386      go build -ldflags="-s -w -X main.gVersion=$version" && zip dist/lf-windows-386.zip            lf.exe --move
-CGO_ENABLED=0 GOOS=windows   GOARCH=amd64    go build -ldflags="-s -w -X main.gVersion=$version" && zip dist/lf-windows-amd64.zip          lf.exe --move
+for params in \
+    "android arm64 lf-android-arm64.tar.gz lf" \
+    "darwin amd64 lf-darwin-amd64.tar.gz lf" \
+    "dragonfly amd64 lf-dragonfly-amd64.tar.gz lf" \
+    "freebsd 386 lf-freebsd-386.tar.gz lf" \
+    "freebsd amd64 lf-freebsd-amd64.tar.gz lf" \
+    "freebsd arm lf-freebsd-arm.tar.gz lf" \
+    "illumos amd64 lf-illumos-amd64.tar.gz lf" \
+    "linux 386 lf-linux-386.tar.gz lf" \
+    "linux amd64 lf-linux-amd64.tar.gz lf" \
+    "linux arm lf-linux-arm.tar.gz lf" \
+    "linux arm64 lf-linux-arm64.tar.gz lf" \
+    "linux ppc64 lf-linux-ppc64.tar.gz lf" \
+    "linux ppc64le lf-linux-ppc64le.tar.gz lf" \
+    "linux mips lf-linux-mips.tar.gz lf" \
+    "linux mipsle lf-linux-mipsle.tar.gz lf" \
+    "linux mips64 lf-linux-mips64.tar.gz lf" \
+    "linux mips64le lf-linux-mips64le.tar.gz lf" \
+    "linux s390x lf-linux-s390x.tar.gz lf" \
+    "netbsd 386 lf-netbsd-386.tar.gz lf" \
+    "netbsd amd64 lf-netbsd-amd64.tar.gz lf" \
+    "netbsd arm lf-netbsd-arm.tar.gz lf" \
+    "openbsd 386 lf-openbsd-386.tar.gz lf" \
+    "openbsd amd64 lf-openbsd-amd64.tar.gz lf" \
+    "openbsd arm lf-openbsd-arm.tar.gz lf" \
+    "openbsd arm64 lf-openbsd-arm64.tar.gz lf" \
+    "solaris amd64 lf-solaris-amd64.tar.gz lf" \
+    "windows 386 lf-windows-386.zip lf.exe" \
+    "windows amd64 lf-windows-amd64.zip lf.exe"\
+    # Unsupported
+    # "aix ppc64 lf-aix-ppc64.tar.gz lf" \
+    # "android 386 lf-android-386.tar.gz lf" \
+    # "android amd64 lf-android-amd64.tar.gz lf" \
+    # "android arm lf-android-arm.tar.gz lf" \
+    # "darwin arm64 lf-darwin-arm64.tar.gz lf" \
+    # "js wasm lf-js-wasm.tar.gz lf" \
+    # "plan9 386 lf-plan9-386.tar.gz lf" \
+    # "plan9 amd64 lf-plan9-amd64.tar.gz lf" \
+    # "plan9 arm lf-plan9-arm.tar.gz lf" \
+do 
+    set -- $params
+    # https://golang.org/doc/install/source#environment
+    CGO_ENABLED=0 GOOS=$1 GOARCH=$2 go build -ldflags="-s -w -X main.gVersion=$version"
+    if test "$?" = "0"; then
+        if echo $3 | grep -E "zip$" >/dev/null; then
+            zip dist/$3 $4 --move || exit 5
```

Comment:
Correct me if I'm wrong, but I think `exit` will exit the entire script instead of moving on the next platform, which is probably not desirable.

Reply:
That was intentional. I thought that if `zip` errors out, something must be really wrong.

Reply:
OK no problem, I was under the impression you meant to continue going through all the other builds if `zip`/`tar` fails.